tracker_test.go

```diff
@@ -6,71 +6,93 @@ import (
 	"github.com/almighty/almighty-core/app"
 	"github.com/almighty/almighty-core/app/test"
 	"github.com/almighty/almighty-core/gormapplication"
+	"github.com/almighty/almighty-core/gormsupport"
 	"github.com/almighty/almighty-core/resource"
 )
 
+type trackerAttr struct {
+	Type string
+	URL  string
+}
+
+func getTrackerPayload(attr trackerAttr) app.CreateTrackerPayload {
+	return app.CreateTrackerPayload{
+		Data: &app.TrackerData{
+			Type: APIStringTypeTracker,
+			Attributes: &app.TrackerAttributes{
+				Type: attr.Type,
+				URL:  attr.URL,
+			},
+		},
+	}
+}
+
 func TestCreateTracker(t *testing.T) {
 	resource.Require(t, resource.Database)
 	controller := TrackerController{Controller: nil, db: gormapplication.NewGormDB(DB), scheduler: RwiScheduler}
-	payload := app.CreateTrackerAlternatePayload{
+
+	payload := getTrackerPayload(trackerAttr{
 		URL:  "http://issues.jboss.com",
 		Type: "jira",
-	}
-
+	})
 	_, created := test.CreateTrackerCreated(t, nil, nil, &controller, &payload)
-	if created.ID == "" {
+	if *created.Data.ID == "" {
 		t.Error("no id")
 	}
 }
 
 func TestGetTracker(t *testing.T) {
 	resource.Require(t, resource.Database)
+	defer gormsupport.DeleteCreatedEntities(DB)()
+
 	controller := TrackerController{Controller: nil, db: gormapplication.NewGormDB(DB), scheduler: RwiScheduler}
-	payload := app.CreateTrackerAlternatePayload{
+	payload := getTrackerPayload(trackerAttr{
 		URL:  "http://issues.jboss.com",
 		Type: "jira",
-	}
+	})
 
 	_, result := test.CreateTrackerCreated(t, nil, nil, &controller, &payload)
-	test.ShowTrackerOK(t, nil, nil, &controller, result.ID)
-	_, tr := test.ShowTrackerOK(t, nil, nil, &controller, result.ID)
+	resultID := *result.Data.ID
+	test.ShowTrackerOK(t, nil, nil, &controller, resultID)
+	_, tr := test.ShowTrackerOK(t, nil, nil, &controller, resultID)
```

Comment:
Did not notice that while changing test case, it was already there before. Will remove first call. thanks @sbose78